### PR TITLE
Dropdown filters

### PR DESCRIPTION
**Pull request by fxstr**

### CODE REVIEW

File: src/components/filterOverlay/BacteriaFilters.js [hunk 1]
```diff
@@ -2,59 +2,35 @@ import React from 'react';
...
-     */
-    itemSelectionChangeHandler(item) {
-        if (this.isFilterSelected(item)) this.props.selectedFilters.removeFilter(item);
-        else this.props.selectedFilters.addFilter(item);
+    toggleBacteria() {
+        this.props.changeDetailPanelContent(filterTypes.bacterium);
     }
 
     @computed get metabolisms() {
         const applications = [
```
Comment (eventEmitter) [inline src/components/filterOverlay/BacteriaFilters.js]: `applications` ==> `metabolisms`

File: src/components/filterOverlay/BacteriaFilters.js [hunk 2]
```diff
@@ -2,59 +2,35 @@ import React from 'react';
...
-            this.props.filterValues.getValuesForProperty('bacterium', 'aerobic')
+            this.props.filterValues.getValuesForProperty(filterTypes.bacterium, 'aerobic')
```
Comment (eventEmitter) [inline src/components/filterOverlay/BacteriaFilters.js]: This line is pretty generic and somewhat fancy. I had to read code in many external source files in order to grasp what is happening here. a simple comment would help here.

File: src/components/filterOverlay/BacteriaFilters.js [hunk 3]
```diff
@@ -2,59 +2,35 @@ import React from 'react';
 import { View, StyleSheet } from 'react-native';
 import { computed } from 'mobx';
 import { observer } from 'mobx-react';
+import { filterTypes } from 'infect-frontend-logic';
 import FilterOverlayTitle from '../filterOverlayTitle/FilterOverlayTitle';
 import FilterOverlaySwitchItem from '../filterOverlaySwitchItem/FilterOverlaySwitchItem';
 import log from '../../helpers/log';
+import FilterList from './FilterList';
```
[the remark this answers is not in the document]
Comment (fxstr) [reply src/components/filterOverlay/BacteriaFilters.js]: I will! Until recently, the Joinbox eslint didn't allow file endings which is why those are missing now. I'll update all extensions whenever I touch any imports.

File: src/components/filterOverlay/BacteriaFilters.js [hunk 4]
```diff
@@ -2,59 +2,35 @@ import React from 'react';
...
 
 @observer
 export default class BacteriaFilters extends React.Component {
 
-    componentDidMount() {
-        log('BacteriaFilters: Mounted');
+    constructor(...props) {
+        super(...props);
+        this.toggleBacteria = this.toggleBacteria.bind(this);
```
[the remark this answers is not in the document]
Comment (fxstr) [reply src/components/filterOverlay/BacteriaFilters.js]: This is a bound event listener, yes. I don't like the pattern very much, either, it's far from sexy. Somehow, we have to bind all listeners, though – we have the following options:

1. Arrow functions in the JSX code: `onClick={(ev) => this.methodName(ev)}`. Advantage: Short and clear. Disadvantage: Function is created every time render method is called; this is only a problem for a) bad code with too many render calls or components with lots of instances. 
1. Bound functions (through constructor) `constructor() { this.methodName = this.methodName.bind(this }`. Disadvantage: Shitty to maintain, lots of code. Advantage: Function is only created once, see https://github.com/airbnb/javascript/tree/master/react: « Bind event handlers for the render method in the constructor»
1. Bind functions through [decorator](https://www.npmjs.com/package/autobind-decorator): `@bound methodName() {}`. Disadvantage: Stage 2 (https://github.com/tc39/proposals), Advantage: We already use decorators for MobX, short, no performance issues. 
1. Use functions as object properties: ´class A { methodName = () => {} }´. Advantage: It looks clean and sharp. Disadvantage: [More than expected](https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1), e.g. missing inheritance if transpiled.

Do you have any favorite solution? I'm fine with using 1 if one knows what he's doing; the future is probably 3.

File: src/components/filterOverlay/BacteriaFilters.js [hunk 5]
```diff
@@ -2,59 +2,35 @@ import React from 'react';
...
     }
 
     @computed get shapes() {
         // Strangely, there's a shape in the DB that doesn't have a value (it's undefined) …
-        return this.props.filterValues.getValuesForProperty('bacterium', 'shape')
+        return this.props.filterValues.getValuesForProperty(filterTypes.bacterium, 'shape')
```
[the remark this answers is not in the document]
Comment (fxstr) [reply src/components/filterOverlay/BacteriaFilters.js]: 👍 Done, will release with the next batch of work I do on infect-frontend-logic.

File: src/components/filterOverlay/BacteriaFilters.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (eventEmitter) [reply src/components/filterOverlay/BacteriaFilters.js]: I see, this is an issue with react :) If airbnb is recommending 2, then that is what we should use.

File: src/components/filterOverlay/BacteriaFilters.js [hunk 6]
```diff
@@ -2,59 +2,35 @@ import React from 'react';
...
             .filter(item => item.value !== undefined);
```
[the remark this answers is not in the document]
Comment (fxstr) [reply src/components/filterOverlay/BacteriaFilters.js]: .

File: src/components/filterOverlay/FilterList.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+        return filteredValues;
+    }
+
+    render() {
+        return (
+            <View style={styles.container}>
+                { this.items.map((item, index) => (
+                    <FilterOverlaySwitchItem
+                        key={item.niceValue}
```
[the remark this answers is not in the document]
Comment (eventEmitter) [reply src/components/filterOverlay/FilterList.js]: i like `humanReadableValue` much more! No need to change this now!

**Pull request merged**